### ISSUE TITLE
Is it possible to support 3D convolution?

### ISSUE THREAD

**Issue by leao1995** [1zb/deformable-convolution-pytorch#8]
I want to use 3D version of deformable convolution, but it seems like your code only support 2D. What change should be made to adapt to 3D convolution? Is it possible to just change the number of offsets?

**Comment by 1zb** [comment]
The forward of `ConvOffset2D` contains two part: `deformable_im2col` and `gemm`. To generalize it to 3D version, the function `deformable_im2col` needs a rewrite (`deformable_im3d2col`).

Similarly, in backward propagation, following functions should be implemented:
1. `deformable_col2im3d_coord`
2. `deformable_col2im3d`

In the near future, I may take a try, but don't expect too much.

**Comment by bkvie** [comment]
What would 3d conv change, isn't the offset applied all layers anyways?

**Comment by jmarsil** [comment]
Has anyone given this a shot yet?

**Comment by KrakenLeaf** [comment]
Hi, you can take a look at: https://github.com/KrakenLeaf/deform_conv_pytorch
Let me know if you find any bugs :)

**Comment by jmarsil** [comment]
@KrakenLeaf awesome, awesome, awesome, I'll integrate this into one of my auto segmentation pipelines and let you know!!

**Comment by salmank255** [comment]
Hi @KrakenLeaf,
The 3d deformable convolution is amazing. What about 3d deformable RoI pooling? Do you have the 3d version of pooling layer? if yes please share it. 

Thanks

**Comment by KrakenLeaf** [comment]
Hi @salmank255 ,
Thanks, but unfortunately I did not implement 3D deformable ROI pooling yet. 
But this is definitely something to look at.

**Comment by salmank255** [comment]
Yeah @KrakenLeaf, I am working on it, but didn't complete it yet. If you convert it, please let me know.
Thanks
